File: GO/Reptiles.go
package main

import "fmt"

// Interface para los métodos abstractos
type Reptiles interface {
	PonerHuevos() string
}

// Struct “base” que simula la clase padre
type BaseReptiles struct {
	Nombre string 
	Piel string
}

// Metodo concreto de BaseReptiles
func (b BaseReptiles) Sangre() {
	fmt.Println("Sangre fria.")
}

// Constructor para BaseReptiles
func NewBaseReptiles(nombre, piel string) *BaseReptiles {
	return &BaseReptiles{
		Nombre: nombre,
		Piel: piel,
	}
}

File: C#/C#/clases_abstractas/Reptiles.cs
using System;

namespace C_.clases_abstractas
{
    internal abstract class Reptiles
    {
        public String nombre;
        public String piel;

        public Reptiles(String nombre, String piel)
        {
            this.nombre = nombre;
            this.piel = piel;
        }

        // Metodo abstracto
        public abstract void ponerHuevos();

        public void sangre()
        {
            Console.WriteLine("Sangre fria.");
        }
    }
}
